src/vcpkg/base/downloads.cpp

```diff
@@ -1152,145 +574,43 @@ namespace vcpkg
-                                          machine_readable_progress,
-                                          fs,
-                                          download_path_part_path,
-                                          *split_uri_view,
-                                          sanitized_url))
-                    {
-                        return DownloadPrognosis::NetworkErrorProxyMightHelp;
-                    }
-
-                    if (!check_downloaded_file_hash(
-                            context, fs, sanitized_url, download_path_part_path, maybe_sha512, out_sha512))
-                    {
-                        return DownloadPrognosis::OtherError;
-                    }
-
-                    fs.rename(download_path_part_path, download_path, VCPKG_LINE_INFO);
-                    return DownloadPrognosis::Success;
-                }
-            }
-        }
-#endif
         // Create directory in advance, otherwise curl will create it in 750 mode on unix style file systems.
         const auto dir = download_path_part_path.parent_path();
         if (!dir.empty())
         {
             fs.create_directories(dir, VCPKG_LINE_INFO);
         }
 
-        auto cmd = Command{"curl"}
-                       .string_arg("--fail")
-                       .string_arg("--retry")
-                       .string_arg("3")
-                       .string_arg("-L")
-                       .string_arg(url_encode_spaces(raw_url))
-                       .string_arg("--create-dirs")
-                       .string_arg("--output")
-                       .string_arg(download_path_part_path);
-        add_curl_headers(cmd, headers);
-        bool seen_any_curl_errors = false;
-        // if seen_any_curl_errors, contains the curl error lines starting with "curl:"
-        // otherwise, contains all curl's output unless it is the machine readable output
-        std::vector<std::string> likely_curl_errors;
-        auto maybe_exit_code = cmd_execute_and_stream_lines(context, cmd, [&](StringView line) {
-            const auto maybe_parsed = try_parse_curl_progress_data(line);
-            if (const auto parsed = maybe_parsed.get())
-            {
-                machine_readable_progress.println(Color::none,
-                                                  LocalizedString::from_raw(fmt::format("{}%", parsed->total_percent)));
-                return;
-            }
+        // Retry on transient errors:
+        // Transient error means either: a timeout, an FTP 4xx response code or an HTTP 408, 429, 500, 502, 503 or
+        // 504 response code. https://everything.curl.dev/usingcurl/downloads/retry.html#retry
+        using namespace std::chrono_literals;
+        static constexpr std::array<std::chrono::seconds, 3> attempt_delays = {0s, 1s, 2s};
+        DownloadPrognosis prognosis = DownloadPrognosis::NetworkErrorProxyMightHelp;
+        for (size_t attempt_count = 0; attempt_count < attempt_delays.size() + 1; attempt_count++)
```

Comment:
Please add check to avoid out of bounds accessing after the third attempt:
```
if (attempt_delays.size() != attempt_count)
{
    std::this_thread::sleep_for(attempt_delays[attempt_count]);
}
```